servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java

```diff
@@ -2191,6 +2191,18 @@ boolean validateRoleCertificateRequest(final String csr, final String domainName
             return false;
         }
 
+        // validate uriHostname in request matches the uriHostname in cert
+        if (!validateUriHostname(certReq.getUriHostname(), cert)) {
+            return false;
+        }
+
+        // validate InstanceID in request matches the instanceid in the cert
+        if (!validateInstanceId(certReq.getInstanceId(), cert)) {
+            // Todo: minitor the logs, and make this a hard failure in later iteration
+            LOGGER.error("unable to match request id: {} with cert id: {}", certReq.getInstanceId(), X509CertUtils.extractRequestInstanceId(cert));
+            return true;
```

Comment:
we should not return true here. otherwise we're skipping the last check.

Reply:
done